builder/proxmox/step_finalize_template_config.go

```diff
@@ -93,6 +94,29 @@ func (s *stepFinalizeTemplateConfig) Run(ctx context.Context, state multistep.St
 		}
 	}
 
+	if c.AdditionalISOFiles != nil {
+		vmParams, err := client.GetVmConfig(vmRef)
+		if err != nil {
+			err := fmt.Errorf("Error fetching template config: %s", err)
+			state.Put("error", err)
+			ui.Error(err.Error())
+			return multistep.ActionHalt
+		}
+		for idx := range c.AdditionalISOFiles {
+			cdrom := c.AdditionalISOFiles[idx].Device + strconv.Itoa(c.AdditionalISOFiles[idx].BusNumber)
+			if c.AdditionalISOFiles[idx].Unmount {
+				if vmParams[cdrom] == nil || !strings.Contains(vmParams[cdrom].(string), "media=cdrom") {
+					err := fmt.Errorf("Cannot eject ISO from cdrom drive, %s is not present or not a cdrom media", cdrom)
+					state.Put("error", err)
+					ui.Error(err.Error())
+					return multistep.ActionHalt
+				}
+				changes[cdrom] = "none,media=cdrom"
+			} else {
+				changes[cdrom] = c.AdditionalISOFiles[idx].Filename + ",media=cdrom"
+			}
```

Comment:
This wouldn't be a change, right? It is what you set already in the StartVM step.